src/main/java/reactor/ipc/netty/http/client/HttpClientWSOperations.java

```diff
@@ -111,8 +111,10 @@ public void onInboundNext(ChannelHandlerContext ctx, Object msg) {
 					}
 				}
 				catch (WebSocketHandshakeException wshe) {
-					onInboundError(wshe);
-					return;
+					if (clientError) {
```

Comment:
Hi Simon,

With the current proposal the CF Java Client is able to negotiate again in case of a token expiration:
```
cloudfoundry-client-nio-5 cloudfoundry-client.request           WS     https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream
cloudfoundry-client-nio-8 cloudfoundry-client.token             Negotiating using token provider
cloudfoundry-client-nio-8 cloudfoundry-client.request           POST   https://login.[REDACTED]:443/oauth/token
cloudfoundry-client-nio-3 cloudfoundry-client.response          200    https://login.[REDACTED]:443/oauth/token (647 ms)
cloudfoundry-client-nio-3 cloudfoundry-client.token             Refresh Token: [REDACTED]
cloudfoundry-client-nio-3 cloudfoundry-client.token             Refresh Token Issued At:  2017-06-30T19:22:07 UTC
cloudfoundry-client-nio-3 cloudfoundry-client.token             Refresh Token Expires At: 2017-07-07T19:22:07 UTC
cloudfoundry-client-nio-3 cloudfoundry-client.token             Access Token: [REDACTED]
cloudfoundry-client-nio-3 cloudfoundry-client.token             Access Token Issued At:  2017-06-30T19:22:07 UTC
cloudfoundry-client-nio-3 cloudfoundry-client.token             Access Token Expires At: 2017-06-30T19:32:07 UTC
cloudfoundry-client-nio-8 cloudfoundry-client.response          101    https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream (1.4 s)
main                      cloudfoundry-client.request           WS     https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream
cloudfoundry-client-nio-8 cloudfoundry-client.request           WS     https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream
cloudfoundry-client-nio-8 cloudfoundry-client.response          401    https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream (576 ms)
cloudfoundry-client-nio-2 cloudfoundry-client.token             Negotiating using refresh token
cloudfoundry-client-nio-2 cloudfoundry-client.request           POST   https://login.[REDACTED]:443/oauth/token
cloudfoundry-client-nio-4 cloudfoundry-client.response          200    https://login.[REDACTED]:443/oauth/token (598 ms)
cloudfoundry-client-nio-4 cloudfoundry-client.token             Refresh Token: [REDACTED]
cloudfoundry-client-nio-4 cloudfoundry-client.token             Refresh Token Issued At:  2017-06-30T19:22:07 UTC
cloudfoundry-client-nio-4 cloudfoundry-client.token             Refresh Token Expires At: 2017-07-07T19:22:07 UTC
cloudfoundry-client-nio-4 cloudfoundry-client.token             Access Token: [REDACTED]
cloudfoundry-client-nio-4 cloudfoundry-client.token             Access Token Issued At:  2017-06-30T19:32:26 UTC
cloudfoundry-client-nio-4 cloudfoundry-client.token             Access Token Expires At: 2017-06-30T19:42:26 UTC
cloudfoundry-client-nio-2 cloudfoundry-client.response          101    https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream (902 ms)
```

If we signal with an error then it will not be able to do another negotiation:
```
cloudfoundry-client-nio-5 cloudfoundry-client.request           WS     https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream
cloudfoundry-client-nio-8 cloudfoundry-client.token             Negotiating using token provider
cloudfoundry-client-nio-8 cloudfoundry-client.request           POST   https://login.[REDACTED]:443/oauth/token
cloudfoundry-client-nio-2 cloudfoundry-client.response          200    https://login.[REDACTED]:443/oauth/token (581 ms)
cloudfoundry-client-nio-2 cloudfoundry-client.token             Refresh Token: [REDACTED]
cloudfoundry-client-nio-2 cloudfoundry-client.token             Refresh Token Issued At:  2017-06-30T19:07:25 UTC
cloudfoundry-client-nio-2 cloudfoundry-client.token             Refresh Token Expires At: 2017-07-07T19:07:25 UTC
cloudfoundry-client-nio-2 cloudfoundry-client.token             Access Token: [REDACTED]
cloudfoundry-client-nio-2 cloudfoundry-client.token             Access Token Issued At:  2017-06-30T19:07:25 UTC
cloudfoundry-client-nio-2 cloudfoundry-client.token             Access Token Expires At: 2017-06-30T19:17:25 UTC
cloudfoundry-client-nio-8 cloudfoundry-client.response          101    https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream (1.4 s)
main                      cloudfoundry-client.request           WS     https://doppler.[REDACTED]:443/apps/a7c44e13-1b9f-46f5-b213-b3b6235a36e1/stream
cloudfoundry-client-nio-4 reactor.ipc.netty.channel.ChannelOperations [HttpClientWS] Error processing connection. Requesting close the channel
io.netty.handler.codec.http.websocketx.WebSocketHandshakeException: Invalid handshake response getStatus: 401 Unauthorized
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker13.verify(WebSocketClientHandshaker13.java:193)
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker.finishHandshake(WebSocketClientHandshaker.java:212)
	at reactor.ipc.netty.http.client.HttpClientWSOperations.onInboundNext(HttpClientWSOperations.java:110)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:125)
```

Regards,
Violeta